vertx-mongo-client/src/main/java/io/vertx/ext/mongo/impl/codec/json/JsonObjectCodec.java

```diff
@@ -106,6 +149,14 @@ protected Object readObjectId(BsonReader reader, DecoderContext ctx) {
 
   @Override
   protected Object readDateTime(BsonReader reader, DecoderContext ctx) {
```

Comment:
Maybe we should just accept Mongo extended JSON as described in their docs PLUS a few other "extras" such as representing dates as a simple $date, long